### PR TITLE
[GLT-3520] Added separate data review for run-libraries

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

File: miso-web/src/main/webapp/scripts/list_runaliquot.js [hunk 1]
```diff
@@ -66,19 +108,37 @@ ListTarget.runaliquot = {
...
+      mRender: function(data, type, full) {
+        if (data === undefined || data === null) {
+          return 'Pending';
+        }
+        var status = data ? 'Pass' : 'Fail';
+        if (type !== 'display') {
+          return status;
+        }
+        return '<div class="tooltip"><span>' + status + '</span>'
+            + '<span class=tooltiptext>Set by ' + full.dataReviewer + ', ' + full.dataReviewDate
```
Comment (avarsava) [inline miso-web/src/main/webapp/scripts/list_runaliquot.js:139]: Inconsistent quotation on class attribute of div and span?

File: miso-web/src/main/webapp/scripts/runLibraryMetrics_ajax.js [hunk 2]
```diff
@@ -65,11 +65,21 @@ var RunLibraryMetrics = (function($) {
         Utils.showOkDialog('Error', ['No changes to save']);
         return;
       }
-      Utils.ajaxWithDialog('Saving...', 'PUT', Urls.rest.qcStatuses.bulkUpdate, changes.map(Utils.array.get('update')), function() {
-        changes.forEach(function(change) {
-          change.rowData.update(change.update);
+      function save() {
+        Utils.ajaxWithDialog('Saving...', 'PUT', Urls.rest.qcStatuses.bulkUpdate, changes.map(Utils.array.get('update')), function() {
+          changes.forEach(function(change) {
+            change.rowData.dataReview = null; // TODO: test
```
Comment (avarsava) [inline miso-web/src/main/webapp/scripts/runLibraryMetrics_ajax.js:71]: Ticket for TODO?

**Review by avarsava — APPROVED** [review]

**Review by callunity — COMMENTED** [review]
I am trying to "Set QC" on library aliquots that I've already QC Passed and Data Reviewed (both on the aliquots and on the run), but it's throwing a JS "showConfirmDialog is not defined" error. I seem to have the most recent version of everything deployed; are you able to replicate this?

File: miso-web/src/main/webapp/scripts/list_runaliquot.js (hunk 1 above)
Comment (djcooke) [reply miso-web/src/main/webapp/scripts/list_runaliquot.js]: Fixed

File: miso-web/src/main/webapp/scripts/runLibraryMetrics_ajax.js (hunk 2 above)
Comment (djcooke) [reply miso-web/src/main/webapp/scripts/runLibraryMetrics_ajax.js]: This was more a note to self that I forgot to remove. Removed now

**Comment by djcooke** [comment]
@callunity I swear I already came across that and fixed it during development! Not sure what happened, but it's fixed again!

**Comment by djcooke** [comment]
This change will be held back until the next release cycle to allow time for SOP changes

**Review by callunity — APPROVED** [review]

**Review by avarsava — APPROVED** [review]

**Review by callunity — APPROVED** [review]

File: miso-web/src/main/webapp/scripts/runLibraryMetrics_ajax.js [hunk 3]
```diff
@@ -43,22 +56,37 @@ var RunLibraryMetrics = (function($) {
...
+          description: 'Pending'
+        }].concat(Constants.runLibraryQcStatuses),
         getLabel: Utils.array.get('description')
       }, {
         label: 'Note',
         property: 'note',
         type: 'text'
       }];
-      Utils.showDialog('Set All Run-Library QCs', 'Set', fields, function(results) {
+      Utils.showDialog('QC All Run-Libraryies', 'Apply', fields, function(results) {
```
[the remark this answers is not in the document]
Comment (djcooke) [reply miso-web/src/main/webapp/scripts/runLibraryMetrics_ajax.js]: Fixed

**Pull request merged**